R/sig_table.R

```diff
@@ -0,0 +1,40 @@
+#' Generate a signature table for a validation report
+#'
+#' @param people data frame with variables: role, name, and title
+#'
+#' @return A data frame with people info formatted for validation report with
+#'   the
+#'
+#' @export
+vt_generate_sig_table <- function(people){
+  # check that the tables have correct variables
+  if (!all(c("role", "name", "title") %in% tolower(names(people)))) {
+    stop(paste0("people table must have variables: role, name, and title. ",
+                "Contains: ", paste0(tolower(names(people)), collapse = ", ")))
+  }
+
+  people$signature <- NA
+  people$date <- NA
+  names(people) <- tolower(names(people))
+  people$name_and_title <- paste(people$name, people$title,  sep = ", " )
+  people[, c("role", "name_and_title", "signature", "date")]
+
+}
+
+#' Return the templated code for inclusion in the validation report
+#'
+#' @param people data frame with variables: role, name, and title
+#' @param template Name of the template to render
+#' @param output directory passed to `render_template()`
+#'
+#' @return The text to include in a validation R markdown document
+#'
+#' @export
+vt_render_sig_table <- function(people, template = "sig_table.Rmd", output) {
+  render_template(template = template,
```

Comment:
should we add ability to add this into the validation.yml (#57 ) be default?

Reply:
this is a more general thought to consider for all templates @elimillera @mariev

Reply:
@thebioengineer Thats what I was thinking. I think the majority of the rendering could be specified in the config and the helper rendering functions would be used where they are specified

Reply:
If we write report rendering function correctly, we should just be able to cycle through each of the files in the config field `validation_files` in order.

My vote that the default report template includes the code to generate the various kable calls, but we should have the flexibility to accomodate a scenario where the parent report only contains the setup chunk and any rmd/md with signature tables, etc is in the first 1+ items of that `validation_files` field. 

One use case I can imagine is that the org has an approved form for capturing signatures which is required by SOP.